openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/FairGBMDescriptorUtil.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2022 Feedzai
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.feedzai.openml.provider.lightgbm;
+
+import com.feedzai.openml.provider.descriptor.ModelParameter;
+import com.feedzai.openml.provider.descriptor.fieldtype.ChoiceFieldType;
+import com.feedzai.openml.provider.descriptor.fieldtype.FreeTextFieldType;
+import com.feedzai.openml.provider.descriptor.fieldtype.NumericFieldType;
+import com.google.common.collect.ImmutableSet;
+
+import com.google.common.collect.Sets;
+import java.util.Set;
+
+/**
+ * Utility to organize all the necessary Machine Learning Hyper-Parameters for configuring the training of LightGBM.
+ *
+ * @author Andre Cruz (andre.cruz@feedzai.com)
+ * @since 1.2.1 // TODO: check what's the current version
+ */
+public class FairGBMDescriptorUtil extends LightGBMDescriptorUtil {
+
+    /**
+     * An alias to ease the readability of parameters' configuration that are not mandatory.
+     */
+    private static final boolean NOT_MANDATORY = false;
+
+    /**
+     * An alias to ease the readability of parameters' configuration that are not mandatory.
+     */
+    private static final boolean MANDATORY = true;
+
+    /**
+     * Defines the set of model parameters accepted by the FairGBM algorithm.
+     *
+     * Parameters are sorted such that commonly changed parameters come first, and parameters that should seldom be
+     * changed come last.
+     */
+    static final Set<ModelParameter> PARAMS = Sets.union(ImmutableSet.of(
+            // The single parameter that will change for every different dataset
+            new ModelParameter(
+                    "constraint_group_column",
+                    "(Fairness) Sensitive group column",
```

Comment:
FYI we're using the "(Fairness)" prefix to group together all FairGBM parameters